logcat/src/nativeMain/kotlin/logcat/extLogcatLogger.kt

```diff
@@ -0,0 +1,36 @@
+package logcat
+
+import kotlin.native.concurrent.AtomicReference
+import kotlin.native.concurrent.freeze
+
+private val loggerReference: AtomicReference<LogcatLogger> = AtomicReference(NoLog)
+private val installedThrowable: AtomicReference<Throwable?> = AtomicReference(null)
+
+@PublishedApi
+internal actual val LogcatLogger.Companion.logger: LogcatLogger
+  get() = loggerReference.value
+
+actual fun LogcatLogger.Companion.install(logger: LogcatLogger) {
+  logger.freeze()
```

Comment:
At this point, freezing should be optional and controlled by the runtime memory setting

Reply:
Is this referring to the new memory management for Kotlin/Native?  Is there a way to check the memory setting at runtime to optionally freeze the logger?